### PR TITLE
KFRD-29 Logging was added

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

File: src/main/scala/com/bwsw/kafka/reader/CheckpointInfoProcessor.scala [hunk 1]
```diff
@@ -29,16 +30,18 @@ import com.bwsw.kafka.reader.entities.{OutputEnvelope, TopicInfoList, TopicParti
...
     * Saves checkpoint data with help of embedded tools of Kafka
     *
     * @param envelopes list of OutputEnvelope entities
     */
   def save(envelopes: List[OutputEnvelope[T]]): Unit = {
+    logger.trace(s"save(envelopes: $envelopes)")
     val partitionsInfo = envelopes.map { envelope =>
       TopicPartitionInfo(envelope.topic, envelope.partition, envelope.offset)
     }
+    logger.debug(s"PartitionInfo list: '$partitionsInfo' based on OutputEnvelope entities: '$envelopes' received" )
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/CheckpointInfoProcessor.scala]: received -> will be committed

File: src/main/scala/com/bwsw/kafka/reader/Consumer.scala [hunk 2]
```diff
@@ -60,6 +60,7 @@ class Consumer[K,V](settings: Consumer.Settings) {
     * @param topicPartitionInfoList entity which contains topics, partitions and offsets.
     */
   def assignWithOffsets(topicPartitionInfoList: TopicPartitionInfoList): Unit = {
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/Consumer.scala:62]: add a log before "consumer.assign" and a log before seek circle

File: src/main/scala/com/bwsw/kafka/reader/Consumer.scala [hunk 3]
```diff
@@ -81,10 +83,12 @@ class Consumer[K,V](settings: Consumer.Settings) {
     * @throws NoSuchElementException if no one of the topics does not exist in Kafka
     */
   def assign(topicInfoList: TopicInfoList): Unit = {
+    logger.trace(s"topicInfoList: $topicInfoList")
     val topicPartitions = covertToTopicPartition(topicInfoList).asJavaCollection
     if (!topicPartitions.isEmpty) {
       consumer.assign(topicPartitions)
     } else {
+      logger.error(s"No one of topics: $topicInfoList does not exist, NoSuchElementException will be thrown")
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/Consumer.scala]: double negation - fix it

Comment (placccebo) [reply src/main/scala/com/bwsw/kafka/reader/Consumer.scala]: in the whole project

File: src/main/scala/com/bwsw/kafka/reader/Consumer.scala [hunk 4]
```diff
@@ -81,10 +83,12 @@ class Consumer[K,V](settings: Consumer.Settings) {
     * @throws NoSuchElementException if no one of the topics does not exist in Kafka
     */
   def assign(topicInfoList: TopicInfoList): Unit = {
+    logger.trace(s"topicInfoList: $topicInfoList")
     val topicPartitions = covertToTopicPartition(topicInfoList).asJavaCollection
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/Consumer.scala]: typo in the word "convert"

File: src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala [hunk 5]
```diff
@@ -42,17 +44,21 @@ class MessageQueue[K,V](consumer: Consumer[K,V]) {
     * @param n count of InputEnvelope entities to extract
     */
   def take(n: Int): List[InputEnvelope[V]] = {
+    logger.trace(s"take(n: $n)")
     if (buffer.size < n) {
+      logger.debug("Count of entities in buffer is not enough, new entities will be received from Kafka")
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala]: received -> retreived

File: src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala [hunk 6]
```diff
@@ -42,17 +44,21 @@ class MessageQueue[K,V](consumer: Consumer[K,V]) {
...
       fill()
     }
 
     val sizeAfterFill = buffer.size
+    logger.debug(s"Count of entities in buffer after receiving from Kafka is: $sizeAfterFill")
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala]: receiving -> retreiving

File: src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala [hunk 7]
```diff
@@ -42,17 +44,21 @@ class MessageQueue[K,V](consumer: Consumer[K,V]) {
...
 
     val envelopes = if (sizeAfterFill < n) {
       buffer.take(sizeAfterFill)
     } else {
       buffer.take(n)
     }
+    logger.debug(s"Entities: $envelopes received")
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala]: The following entities: $envelopes will be returned

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/main/scala/com/bwsw/kafka/reader/Consumer.scala (hunk 2 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/Consumer.scala:62]: ok

File: src/main/scala/com/bwsw/kafka/reader/CheckpointInfoProcessor.scala (hunk 1 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/CheckpointInfoProcessor.scala]: ok

File: src/main/scala/com/bwsw/kafka/reader/Consumer.scala (hunk 4 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/Consumer.scala]: ok

File: src/main/scala/com/bwsw/kafka/reader/Consumer.scala (hunk 3 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/Consumer.scala]: ok

File: src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala (hunk 5 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala]: retrieved*, ok

File: src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala (hunk 6 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala]: ok

File: src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala (hunk 7 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala]: ok

File: src/main/scala/com/bwsw/kafka/reader/Consumer.scala [hunk 8]
```diff
@@ -60,69 +60,90 @@ class Consumer[K,V](settings: Consumer.Settings) {
...
-      topicPartitionsWithOffsets.map {
-        case (topicPartitions, offset) => topicPartitions
-      }.asJavaCollection
-    )
+
+    val topicPartitions = topicPartitionsWithOffsets.map {
+      case (partition, offset) => partition
+    }
+
+    logger.debug(s"assign topic partitions: $topicPartitions")
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/Consumer.scala]: the topic partitions

File: src/main/scala/com/bwsw/kafka/reader/Consumer.scala [hunk 9]
```diff
@@ -60,69 +60,90 @@ class Consumer[K,V](settings: Consumer.Settings) {
...
   def assign(topicInfoList: TopicInfoList): Unit = {
-    val topicPartitions = covertToTopicPartition(topicInfoList)
+    logger.trace(s"topicInfoList: $topicInfoList")
+    val topicPartitions = convertToTopicPartition(topicInfoList)
     if (topicPartitions.nonEmpty) {
+      logger.debug(s"assign topic partitions: $topicPartitions")
       consumer.assign(topicPartitions.asJavaCollection)
     } else {
-      throw new NoSuchElementException(s"No one of topics: $topicInfoList does not exist")
+      logger.error(s"No one of topics: $topicInfoList exists, NoSuchElementException will be thrown")
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/Consumer.scala]: you should define the uniform format of logs: the uppercase or the lowercase first letter

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/main/scala/com/bwsw/kafka/reader/Consumer.scala (hunk 8 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/Consumer.scala]: ok

File: src/main/scala/com/bwsw/kafka/reader/Consumer.scala (hunk 9 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/Consumer.scala]: Uppercase, excluding messages for "trace" log level

**Comment by MedvedevBW** [comment]
@placccebo

**Review by placccebo — APPROVED** [review]

**Pull request merged**